### PR TITLE
SCM-960: improved text-search for complex labels

### PR DESCRIPTION
**Pull request by mean2me**
Added a "decode" function as a required PropType (default is provided) to customize text search in case label is not a simple string.

### CODE REVIEW

File: Dropdown/src/Dropdown.jsx [hunk 1]
```diff
@@ -171,30 +171,36 @@ class Dropdown extends Component {
...
                 closestValueLength = labelToMatch.match(stringToMatchBeginning).length;
                 closestValue = option.value;
                 itemIndex = index;
-
             }
         });
 
         if (closestValue === null) {
-            this.props.options.forEach((option, index) => {
+            props.options.forEach((option, index) => {
```
Comment (tpluscode) [inline Dropdown/src/Dropdown.jsx]: Does this `forEach` look very similar to the one above? Maybe they could be extracted to a function?

File: Dropdown/src/Dropdown.jsx [hunk 2]
```diff
@@ -171,30 +171,36 @@ class Dropdown extends Component {
...
+
+        const { props } = this;
+
         let closestValueLength = 0, closestValue = null, itemIndex = 0;
 
-        this.props.options.forEach((option, index) => {
+        props.options.forEach((option, index) => {
             let regex = this.state.dropdownText.replace(/([.?*+^$[\]\\(){}|-])/g, "\\$1");
 
+            const label = props.decode ? props.decode(option.label) : option.label;
```
Comment (tpluscode) [inline Dropdown/src/Dropdown.jsx]: Is the check necessary? `decode` is marked as required and has a default value. And what will React do if it is assigned null/undefined?

File: Dropdown/src/Dropdown.jsx [hunk 3]
```diff
@@ -412,7 +415,8 @@ Dropdown.defaultProps = {
     autoHide: true,
     className: "",
     isDropDownOpen: false,
-    selectedIndex: -1
+    selectedIndex: -1,
+    decode:(label) => label
```
Comment (tpluscode) [inline Dropdown/src/Dropdown.jsx]: I'm not sure about the name `decode`. Maybe it could be more descriptive. Like `getSearchText`?

**Comment by tpluscode** [comment]
Could we also see a test for keyboard search like with the `<Flag>` case?

**Comment by mean2me** [comment]
That case is covered by the last test added. When we type something, we set "value" property and a first search is performed. Last test simulate the same condition.

**Pull request merged**